docker-entrypoint.sh

```diff
@@ -54,60 +54,58 @@ if [[ ! -f /db/init_done ]] ; then
-                && /app/bin/update_overpass.sh "-O /db/planet.osm.bz2" \
-                && /app/bin/osm3s_query --progress --rules --db-dir=/db/db < /db/db/rules/areas.osm3s \
-                && touch /db/init_done \
-                && rm /db/planet.osm.bz2 \
-                && chown -R overpass:overpass /db \
-                && echo "Overpass ready, you can start your container with docker start" \
-                && exit
-              ) || (
-                echo "Failed to process planet file"
-                exit
-              )
-              ;;
-            403)
-              echo "Access denied when downloading planet file. Check your OVERPASS_PLANET_URL and OVERPASS_COOKIE_JAR_CONTENTS or USE_OAUTH_COOKIE_CLIENT"
-              cat /db/cookie.jar
-              exit
-              ;;
-            *)
-              echo "Failed to download planet file. HTTP status code: ${CURL_STATUS_CODE}"
-              cat /db/planet.osm.bz2
-              exit
-              ;;
-          esac
-          exit
+        CURL_STATUS_CODE=$(curl -L -b /db/cookie.jar -o /db/planet.osm.bz2 -w "%{http_code}" "${OVERPASS_PLANET_URL}")
+        # try again until it's allowed
+        while [ $CURL_STATUS_CODE = "429" ] ; do
+            echo "Server responded with 429 Too many requests. Trying again in 5 minutes..."
+            sleep 300
+            CURL_STATUS_CODE=$(curl -L -b /db/cookie.jar -o /db/planet.osm.bz2 -w "%{http_code}" "${OVERPASS_PLANET_URL}")
         done
+          # for `file:///` scheme curl returns `000` HTTP status code
+        if [[ $CURL_STATUS_CODE = "200" || $CURL_STATUS_CODE = "000" ]]; then
+            (
+              if [[ ! -z "${OVERPASS_PLANET_PREPROCESS+x}" ]]; then
+                  echo "Running preprocessing command: ${OVERPASS_PLANET_PREPROCESS}"
+                  eval "${OVERPASS_PLANET_PREPROCESS}"
+              fi \
+              && /app/bin/init_osm3s.sh /db/planet.osm.bz2 /db/db /app "${META}" "--version=$(osmium fileinfo -e -g data.timestamp.last /db/planet.osm.bz2) --compression-method=${OVERPASS_COMPRESSION} --map-compression-method=${OVERPASS_COMPRESSION} --flush-size=${OVERPASS_FLUSH_SIZE}" \
+              && echo "Database created. Now updating it." \
+              && cp -r /app/etc/rules /db/db \
+              && chown -R overpass:overpass /db \
+              && echo "Updating" \
+              && /app/bin/update_overpass.sh "-O /db/planet.osm.bz2" \
+              && if [[ "OVERPASS_USE_AREAS" = "true" ]]; then
+                   echo "Generating areas..." && /app/bin/osm3s_query --progress --rules --db-dir=/db/db < /db/db/rules/areas.osm3s
+                 fi \
+              && touch /db/init_done \
+              && rm /db/planet.osm.bz2 \
+              && chown -R overpass:overpass /db \
+              && echo "Overpass container ready to receive requests"
+            ) || (
+              echo "Failed to process planet file"
+              exit 1
```

Comment:
I changed the exit codes to 1, I think a plain `exit` will exit with the exit code of the last command which is `echo` and 0.